### DISCUSSION TITLE
Strange 404 request on Production

### DISCUSSION THREAD

**Discussion by tminhduc2811** [vercel/next.js#17569] [Help] [answered]
Hello, I'm having some troubles while running Nextjs in Production
My routes look like 
![image](https://user-images.githubusercontent.com/40265306/94951987-690c6380-050f-11eb-80cb-e235b8e8be56.png)
I use Static Site Generation for all the pages.
When I load the category page, it requests to [category].json which is not existed
![image](https://user-images.githubusercontent.com/40265306/94952149-b4267680-050f-11eb-83b3-42e368f19c00.png)
****
Is there anything wrong with my configuration when using dynamic routes
Thanks.

**Comment by jamesmosier** [comment, marked as answer]
Hi @tminhduc2811! 

1. Do you have a `next/link` (`<Link>`) component that contains a non-valid url on your page?
2. Could you share the contents of your category page that is showing this error? In particular your `getStaticProps` and/or `getStaticPaths` functions

**Reply by tminhduc2811** [reply]
Hello @jamesmosier,
Can you take a look at my repo https://github.com/tminhduc2811/CurtainStoreV2/blob/master/pages/danh-muc/%5Bcategory%5D/index.jsx

This happens only on production, the page still works fine. However, I don't know why it requests to [category].json 
![image](https://user-images.githubusercontent.com/40265306/94954287-1c2a8c00-0513-11eb-8d43-1b3dd12f954c.png)

**Reply by jamesmosier** [reply]
`[category].json` is the result of `getStaticProps` when you navigate to that page via `next/link`. Next prefetches the pages/data found in `<Link href="">`

It looks like you are passing incorrect values to the `<Link` component, likely [here](https://github.com/tminhduc2811/CurtainStoreV2/blob/5874eced8532dca68207017783ae424a4be2f623/components/MenuLink.jsx)

Here is 1 example:

![Screen Shot 2020-10-02 at 2 14 25 PM](https://user-images.githubusercontent.com/2854919/94956146-a8a86080-04b9-11eb-918c-32fb74694173.png)

**Reply by tminhduc2811** [reply]
Oh my bad. I didn't notice that.
Thank you so much @jamesmosier for helping me.
